common/server.cpp

```diff
@@ -168,6 +170,15 @@ AuthenticationResult Server::loginUser(Server_ProtocolHandler *session, QString
     event.mutable_user_info()->CopyFrom(session->copyUserInfo(true, true, true));
     locker.unlock();
     
+    // check if client id exists (older client compatibility)
+    if (clientid.isEmpty()){
+        // client id is empty, either out dated client or client has been modified
+    }
+    else {
+        // update users database table with client id
+        databaseInterface->updateUsersClientID(name, clientid);
```

Comment:
We should move to having one post-login call that updates all the "most recent" fields; this will increase DB load a bit for each connection.